### PR TITLE
added additional logging and fixed some misplaced logging in default/legacy accessors & credential cache.

### PR DESCRIPTION
**Pull request by NerevarineRule**
## Proposed changes

Trying to add more logging in the legacy & default accessor logic and credential cache logic as well.
By doing so we can better identify how credentials get filtered out by the underlying logic.
Also there were a couple ares where logging was not placed in the correct place (ex: getTokenWithEnvironment). so also fixing that. 

## Type of change

- [ ] Feature work
- [ ] Bug fix
- [ ] Documentation
- [x] Engineering change
- [ ] Test
- [ ] Logging/Telemetry

## Risk

- [ ] High – Errors could cause MAJOR regression of many scenarios. (Example: new large features or high level infrastructure changes)
- [ ] Medium – Errors could cause regression of 1 or more scenarios. (Example: somewhat complex bug fixes, small new features)
- [x] Small – No issues are expected. (Example: Very small bug fixes, string changes, or configuration settings changes)

### CODE REVIEW

File: IdentityCore/src/cache/token/MSIDCredentialCacheItem.m [hunk 1]
```diff
@@ -315,25 +315,25 @@ - (BOOL)matchesWithRealm:(nullable NSString *)realm
 {
     if (realm && ![self.realm.msidNormalizedString isEqualToString:realm.msidNormalizedString])
     {
-        MSID_LOG_WITH_CTX(MSIDLogLevelVerbose, nil, @"(%@) cached item did not have a valid realm. Actual: %@, expected: %@", NSStringFromClass(self.class), self.realm, realm);
+        MSID_LOG_WITH_CTX(MSIDLogLevelInfo, nil, @"(%@) cached item did not have a valid realm. Actual: %@, expected: %@", NSStringFromClass(self.class), self.realm, realm);
```
Comment (kaisong1990) [inline IdentityCore/src/cache/token/MSIDCredentialCacheItem.m:318]: Can you double-check if it is ok to use Info level from security perspective?

**Review by kaisong1990 — APPROVED** [review]
Approved with comments

**Review by oldalton — CHANGES_REQUESTED** [review]
Unfortunately, those logs are too verbose to be info level. Also, please make sure you don't log PII (UPN).

File: IdentityCore/src/cache/accessor/MSIDLegacyTokenCacheAccessor.m [hunk 2]
```diff
@@ -260,34 +260,40 @@ - (BOOL)clearWithContext:(id<MSIDRequestContext>)context
...
         }
         if (tokenCacheItem.credentialType == MSIDPrimaryRefreshTokenType)
         {
+            MSID_LOG_WITH_CTX(MSIDLogLevelVerbose, context, @"(Legacy accessor) Credential type is PriamryRefreshToken.");
             return YES;
         }
         
         if (accountIdentifier.homeAccountId && ![tokenCacheItem.homeAccountId isEqualToString:accountIdentifier.homeAccountId])
         {
+            MSID_LOG_WITH_CTX_PII(MSIDLogLevelVerbose, context, @"(Legacy accessor) cached home account id mismatches account identifier value. cached: %@, actual: %@",MSID_PII_LOG_TRACKABLE(tokenCacheItem.homeAccountId),MSID_PII_LOG_TRACKABLE(accountIdentifier.homeAccountId));
```
Comment (oldalton) [inline IdentityCore/src/cache/accessor/MSIDLegacyTokenCacheAccessor.m:274]: Please use pre-calculated homeAccountId hash instead.

File: IdentityCore/src/cache/accessor/MSIDLegacyTokenCacheAccessor.m [hunk 3]
```diff
@@ -260,34 +260,40 @@ - (BOOL)clearWithContext:(id<MSIDRequestContext>)context
     BOOL (^filterBlock)(MSIDCredentialCacheItem *tokenCacheItem) = ^BOOL(MSIDCredentialCacheItem *tokenCacheItem) {
         if ([environmentAliases count] && ![tokenCacheItem.environment msidIsEquivalentWithAnyAlias:environmentAliases])
         {
+            MSID_LOG_WITH_CTX(MSIDLogLevelVerbose, context, @"(Legacy accessor) cached account environment not found in the environment alias list.");
             return NO;
         }
         if (tokenCacheItem.credentialType == MSIDPrimaryRefreshTokenType)
         {
+            MSID_LOG_WITH_CTX(MSIDLogLevelVerbose, context, @"(Legacy accessor) Credential type is PriamryRefreshToken.");
```
Comment (oldalton) [inline IdentityCore/src/cache/accessor/MSIDLegacyTokenCacheAccessor.m:268]: Typo: PriamryRefreshToken.

File: IdentityCore/src/cache/accessor/MSIDLegacyTokenCacheAccessor.m [hunk 4]
```diff
@@ -248,7 +248,7 @@ - (BOOL)clearWithContext:(id<MSIDRequestContext>)context
                                           context:(id<MSIDRequestContext>)context
                                             error:(NSError **)error
 {
-    MSID_LOG_WITH_CTX_PII(MSIDLogLevelVerbose, context, @"(Legacy accessor) Get accounts with environment %@, clientId %@, familyId %@, account identifier %@, legacy identifier %@", authority.environment, clientId, familyId, accountIdentifier.maskedHomeAccountId, accountIdentifier.maskedDisplayableId);
+    MSID_LOG_WITH_CTX_PII(MSIDLogLevelInfo, context, @"(Legacy accessor) Get accounts with environment %@, clientId %@, familyId %@, account identifier %@, legacy identifier %@", authority.environment, clientId, familyId, accountIdentifier.maskedHomeAccountId, accountIdentifier.maskedDisplayableId);
```
Comment (oldalton) [inline IdentityCore/src/cache/accessor/MSIDLegacyTokenCacheAccessor.m:251]: Can you clarify why this change is needed?

File: IdentityCore/src/cache/accessor/MSIDDefaultTokenCacheAccessor.m [hunk 5]
```diff
@@ -1014,6 +1031,8 @@ - (MSIDBaseToken *)getTokenWithEnvironment:(NSString *)environment
         }
     }
     
+    MSID_LOG_WITH_CTX(MSIDLogLevelInfo, context, @"(Default accessor) Found %lu tokens with environment", (unsigned long)[resultTokens count]);
```
Comment (oldalton) [inline IdentityCore/src/cache/accessor/MSIDDefaultTokenCacheAccessor.m:1034]: 👍

File: IdentityCore/src/cache/accessor/MSIDDefaultTokenCacheAccessor.m [hunk 6]
```diff
@@ -997,13 +1013,14 @@ - (MSIDBaseToken *)getTokenWithEnvironment:(NSString *)environment
     }
 
     NSMutableArray<MSIDBaseToken *> *resultTokens = [NSMutableArray new];
+    
+    MSID_LOG_WITH_CTX(MSIDLogLevelInfo, context, @"(Default accessor) Found %lu cached credentials", (unsigned long)[cacheItems count]);
```
Comment (oldalton) [inline IdentityCore/src/cache/accessor/MSIDDefaultTokenCacheAccessor.m:1017]: Is this needed? We already log number below at line 1034.

File: IdentityCore/src/cache/accessor/MSIDDefaultTokenCacheAccessor.m [hunk 7]
```diff
@@ -927,6 +942,7 @@ - (NSString *)homeAccountIdForLegacyId:(NSString *)legacyAccountId
 
     if ([accountCacheItems count])
     {
+        MSID_LOG_WITH_CTX(MSIDLogLevelVerbose, context, @"(Default accessor) found %ld accounts with legacy account id. returning first account.",accountCacheItems.count);
```
Comment (oldalton) [inline IdentityCore/src/cache/accessor/MSIDDefaultTokenCacheAccessor.m:945]: nit: should you cast accountCacheItems.count?

File: IdentityCore/src/cache/accessor/MSIDDefaultTokenCacheAccessor.m [hunk 8]
```diff
@@ -914,6 +927,8 @@ - (NSString *)homeAccountIdForLegacyId:(NSString *)legacyAccountId
                                context:(id<MSIDRequestContext>)context
                                  error:(NSError **)error
 {
+    
+    MSID_LOG_WITH_CTX(MSIDLogLevelInfo, context, @"(Default accessor) querying account with legacy account id");
```
Comment (oldalton) [inline IdentityCore/src/cache/accessor/MSIDDefaultTokenCacheAccessor.m:931]: Does this give enough extra information to justify info-level log?

File: IdentityCore/src/cache/accessor/MSIDDefaultTokenCacheAccessor.m [hunk 9]
```diff
@@ -552,6 +562,8 @@ - (MSIDAccount *)getAccountForIdentifier:(MSIDAccountIdentifier *)accountIdentif
     }
     
     MSIDAccount *firstAccount = nil;
+    
+    MSID_LOG_WITH_CTX(MSIDLogLevelVerbose, context, @"(Default accessor) %ld accounts found with account identifier and authority.",accountCacheItems.count);
```
Comment (oldalton) [inline IdentityCore/src/cache/accessor/MSIDDefaultTokenCacheAccessor.m:566]: Do you need to cast accountCacheItems.count here?

File: IdentityCore/src/cache/accessor/MSIDDefaultTokenCacheAccessor.m [hunk 10]
```diff
@@ -494,6 +502,8 @@ - (BOOL)removeAccessToken:(MSIDAccessToken *)token
         [MSIDTelemetry stopFailedCacheEvent:event wipeData:[_accountCredentialCache wipeInfoWithContext:context error:&wipeError] context:context];
         if (wipeError) MSID_LOG_WITH_CTX_PII(MSIDLogLevelWarning, context, @"Failed to read wipe info with error %@", MSID_PII_LOG_MASKABLE(wipeError));
     }
+    
+    MSID_LOG_WITH_CTX(MSIDLogLevelInfo, context, @"(Default accessor) number of filtered (no-return) accounts: %ld.",noReturnAccountUPNSet.count);
```
Comment (oldalton) [inline IdentityCore/src/cache/accessor/MSIDDefaultTokenCacheAccessor.m:506]: Does this need to be info-level?

File: IdentityCore/src/cache/accessor/MSIDDefaultTokenCacheAccessor.m [hunk 11]
```diff
@@ -433,11 +432,17 @@ - (BOOL)removeAccessToken:(MSIDAccessToken *)token
...
+    {
+        MSID_LOG_WITH_CTX(MSIDLogLevelVerbose, context, @"(Default accessor) found %ld accounts before additional filtering",  (long)allAccounts.count);
+    }
+    
     // we only need it if returning signed in accounts only
     NSSet<NSString *> *accountIdsFromRT = nil;
     NSError *localError;
     if (signedInAccountsOnly)
     {
+        MSID_LOG_WITH_CTX(MSIDLogLevelInfo, context, @"(Default accessor) signedInAccountsOnly Enabled. retrieve account IDs from RT.");
```
Comment (oldalton) [inline IdentityCore/src/cache/accessor/MSIDDefaultTokenCacheAccessor.m:445]: Does this need to be info level?

**Review by oldalton — CHANGES_REQUESTED** [review]
Added a few more questions.

**Comment by oldalton** [comment]
@NerevarineRule, can we close this one or it's still needed?

**Pull request closed without merging**